pkg/virt-controller/watch/pod_test.go

```diff
@@ -190,32 +208,32 @@ var _ = Describe("Pod", func() {
 					ghttp.VerifyRequest("GET", "/api/v1/nodes/targetNode"),
 					ghttp.RespondWithJSONEncoded(http.StatusOK, targetNode),
 				),
+
+				ghttp.CombineHandlers(
+					ghttp.VerifyRequest("POST", "/api/v1/namespaces/default/pods"),
+					//TODO: Validate that posted Pod request is sane
+					ghttp.RespondWithJSONEncoded(http.StatusOK, pod),
+				),
+
 				ghttp.CombineHandlers(
 					ghttp.VerifyRequest("PUT", "/apis/kubevirt.io/v1alpha1/namespaces/default/vms/testvm"),
 					ghttp.VerifyJSONRepresenting(vmInMigrationState),
 					ghttp.RespondWithJSONEncoded(http.StatusOK, vmWithMigrationNodeName),
 				),
 			)
 
-			mockVMService.EXPECT().GetMigrationJob(gomock.Any()).Return(nil, false, nil)
-			mockVMService.EXPECT().StartMigration(gomock.Any(), gomock.Any(), &srcNode, &targetNode, gomock.Any()).Return(nil)
-
-			// Tell the controller that there is a new running Pod
-			lw.Add(pod)
-
-			// Wait until we have processed the added item
-			podController.WaitForSync(stopChan)
-			podController.ShutDownQueue()
-			podController.WaitUntilDone()
+			queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+			key, _ := cache.MetaNamespaceKeyFunc(pod)
+			podCache.Add(pod)
+			queue.Add(key)
+			dispatch.Execute(podCache, queue, key)
 
-			Expect(len(server.ReceivedRequests())).To(Equal(5))
+			Expect(len(server.ReceivedRequests())).To(Equal(7))
```

Comment:
Yeah, but there is no way to get a count of the registered handlers, so the logic here is as simple as we can get.